packages/app/src/sandbox/eval/transpilers/react/refresh-transpiler.ts

```diff
@@ -121,67 +121,69 @@ var registerExportsForReactRefresh = (moduleExports, moduleID) => {
   }
 };
 
+function prelude(module) {
+    window.$RefreshReg$ = (type, id) => {
+        // Note module.id is webpack-specific, this may vary in other bundlers
+        const fullId = module.id + ' ' + id;
+        Refresh.register(type, fullId);
+    }
+    
+    window.$RefreshSig$ = Refresh.createSignatureFunctionForTransform;
+}
+
+function postlude(module) {
+    const isHotUpdate = !!module.hot.data;
+    const prevExports = isHotUpdate ? module.hot.data.prevExports : null;
+    if (isReactRefreshBoundary) {
+        if (isReactRefreshBoundary(module.exports)) {
+            registerExportsForReactRefresh(module.exports, module.id)
+            const currentExports = { ...module.exports };
+
+            module.hot.dispose(function hotDisposeCallback(data) {
+                data.prevExports = currentExports;
+            });
+
+            if (isHotUpdate && shouldInvalidateReactRefreshBoundary(prevExports, currentExports)) {
+                module.hot.invalidate();
+            } else {
+                module.hot.accept();
+            }
+
+            enqueueUpdate();
+        } else if (isHotUpdate && isReactRefreshBoundary(prevExports)) {
+            module.hot.invalidate();
+        }
+    }
+}
+
 module.exports = {
-  enqueueUpdate: debounce(enqueueUpdate, 30),
+  enqueueUpdate,
   isReactRefreshBoundary,
   registerExportsForReactRefresh,
-  shouldInvalidateReactRefreshBoundary
+  shouldInvalidateReactRefreshBoundary,
+  prelude,
+  postlude,
 };
 `.trim();
 
-// const getWrapperCode = (sourceCode: string) =>
```

Comment:
Ah, so you moved the wrapper code to the prelude and the postlude?

Reply:
Yes, that's pretty much it. There's some other small tweaks to readability but no logic changes besides that